typescript/types.ts

```diff
@@ -0,0 +1,37 @@
+export interface Board {
+    width: number,
+    height: number,
+    types: any[],
+    orbs: any[],
+    attic?: any,
+    generateOrbs(): void,
+    evaluate(): MatchData,
+    needsShuffle(): boolean,
+    hasPotentialMatch(): boolean,
+    hasMatch(): boolean,
+    unmatch(): void,
+    shuffle(): void
+}
+
+export type MatchData = [any, number][]
+
+export interface BlanksBelow {
+	[coord: string]: number
+}
+
+export type Coord = [number, number]
+
+export interface PositionInfo {
+	position: {
+		first: Coord,
+		last: Coord
+	}
+}
+
+export type IterchunksWithPosition = [
+	[
+		any[],
+		any[],
+		PositionInfo
+	][]
+]
```

Comment:
There will be much more of this type of specificity as refactoring happens, but this was one of the harder things to type out. This still isn't completely correct, but works (aka doesn't throw and error). The issue is that there could be any number of those `any[]` lists before the `PositionInfo` object. I also asked this [question](http://stackoverflow.com/questions/38237514/declare-an-unknown-number-of-lists-in-a-type-alias).